app/src/main/java/com/mybus/dao/RealmDao.java

```diff
@@ -0,0 +1,176 @@
+package com.mybus.dao;
+
+import android.content.Context;
+
+import com.mybus.model.FavoriteLocation;
+import com.mybus.model.UsageTrackable;
+
+import java.util.Date;
+
+import io.realm.Realm;
+import io.realm.RealmConfiguration;
+import io.realm.RealmObject;
+import io.realm.RealmResults;
+
+/**
+ * Created by Julian Gonzalez <jgonzalez@devspark.com>
+ */
+public abstract class RealmDao<T extends RealmObject> {
+    private Realm mRealm;
+    private Class<T> mType;
+
+    protected RealmDao(Context c, Class<T> clazz) {
+        RealmConfiguration realConfig = new RealmConfiguration.Builder(c)
+                .deleteRealmIfMigrationNeeded()
+                .build();
+        this.mRealm = Realm.getInstance(realConfig);
+        this.mType = clazz;
+    }
+
+    /**
+     * Save or update the given item depends if it already exists or not.
+     * @param item
+     * @return The item updated or null if an error occurred
+     */
+    public T saveOrUpdate(T item) {
+        if (item == null) {
+            return null;
+        }
+        try {
+            mRealm.beginTransaction();
+            T updated = mRealm.copyToRealmOrUpdate(item);
+            mRealm.commitTransaction();
+            return updated;
+        } catch (IllegalArgumentException e) {
+            return null;
+        }
+    }
+
+    /**
+     * Remove an item from realm
+     * @param id To find the item
+     * @return
+     */
+    public boolean remove(Long id) {
+        T item = mRealm.where(mType).equalTo("id", id).findFirst();
+        if (item != null) {
+            mRealm.beginTransaction();
+            item.deleteFromRealm();
+            mRealm.commitTransaction();
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * @return all items on realm
+     */
+    public RealmResults<T> getAll() {
+        return mRealm.where(mType).findAll();
+    }
+
+    /**
+     * Increment the usage count for a given item.
+     * @param id
+     */
+    public void updateItemUsageCount(Long id) {
+        if (!(UsageTrackable.class).isAssignableFrom(mType)) {
+            return;
+        }
+        UsageTrackable item = (UsageTrackable) mRealm.where(mType).equalTo("id", id).findFirst();
```

Comment:
GetBy* uses copyFromRealm:
Which makes a standalone in-memory copy of an already persisted {@link RealmObject}. This is a deep copy that will copy
     * all referenced objects up to the defined depth.
     * The copied object(s) are all detached from Realm so they will no longer be automatically updated. This means
     * that the copied objects might contain data that are no longer consistent with other managed Realm objects.

When you need to update/remove a item you need an attached copy.